app/backend/src/couchers/servicers/auth.py

```diff
@@ -346,7 +346,7 @@ def Authenticate(self, request, context):
                 logger.debug(f"Didn't find user")
                 # do about as much work as if the user was found, reduces timing based username enumeration attacks
                 hash_password(request.password)
-                context.abort(grpc.StatusCode.UNAUTHENTICATED, errors.INVALID_USERNAME_OR_PASSWORD)
+                context.abort(grpc.StatusCode.NOT_FOUND, errors.INVALID_USERNAME_OR_PASSWORD)
```

Comment:
Remove the time delay as that is useless now when we return a different error anyway.

Also change the error message sent to the client to "User not found" or similar.

Reply:
I think the error message and time delay is good, we should just make the other error above also NOT_FOUND (this is needed anyway to fix #531). So no matter if the user doesn't exist or the password is incorrect, you get NOT_FOUND "Invalid username or password"

(Although an attacker can easily know if a user exists with the other login function which tell you how exactly that.)

Reply:
Changed second error message. This should be good now

Out of curiosity, where's the time delay? I'm not noticing it at all

Reply:
@ShmuelTreiger: the `hash_password(...)` call takes some time (not so noticeable as a user, but can be timed easily if you look at how long the request takes). If it always takes longer when the user exists, then it's possible to find usernames by just checking whether the rpc returned quickly or took some time. It's good to not leak that kind of info all over the place unknowingly.

Reply:
OK, I understand the logic, but I don't see where that happens?

Reply:
Just above, on line 348. We hash the password but throw away the result.